js/bootstrap-select.js

```diff
@@ -558,16 +557,16 @@
             inline = this.style.cssText,
             text = $this.data('content') ? $this.data('content') : $this.html(),
             tokens = $this.data('tokens') ? $this.data('tokens') : null,
-            subtext = typeof $this.data('subtext') !== 'undefined' ? '<small class="text-muted">' + $this.data('subtext') + '</small>' : '',
-            icon = typeof $this.data('icon') !== 'undefined' ? '<span class="' + that.options.iconBase + ' ' + $this.data('icon') + '"></span> ' : '',
+            subtext = $this.data('subtext') !== undefined ? '<small class="text-muted">' + $this.data('subtext') + '</small>' : '',
+            icon = $this.data('icon') !== undefined ? '<span class="' + that.options.iconBase + ' ' + $this.data('icon') + '"></span> ' : '',
             isOptgroup = this.parentNode.tagName === 'OPTGROUP',
             isDisabled = this.disabled || (isOptgroup && this.parentNode.disabled);
 
         if (icon !== '' && isDisabled) {
           icon = '<span>' + icon + '</span>';
         }
 
-        if (that.options.hideDisabled && (isDisabled && !isOptgroup || this.parentNode.disabled && isOptgroup)) {
+        if (that.options.hideDisabled && isDisabled) {
```

Comment:
This line needs to be `that.options.hideDisabled && (isDisabled && !isOptgroup || this.parentNode.disabled && isOptgroup)`. With just `that.options.hideDisabled && isDisabled`, if an optgroup has a disabled option in it, the optgroup's header is hidden. I guess we could make a variable called something like `isOptgroupDisabled` and set it equal to `this.parentNode.disabled && isOptgroup`, and then set the if statement to `if (that.options.hideDisabled && (isDisabled && !isOptgroup || isOptgroupDisabled))`.